plugins/templates/create.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const hoek = require('hoek');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/templates',
```

Comment:
Not commenting on the `GET` as it's not part of this PR. 
For the `CREATE`, if we choose to do `PUT /templates/:name/:version` instead of `POST /templates`, if the `name` in the `param` is not the same as the `name` in the `payload`, what is the expected behavior? Do we throw an error? Do we favor the name in the `param`, or the `payload`? We need to make this clear to the users. Same question for `version`.